Services/TodoService.cs

```diff
@@ -3,21 +3,44 @@
 using System.Threading.Tasks;
 using Microsoft.EntityFrameworkCore;
 using Google.Protobuf.WellKnownTypes;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Logging;
 
 using Aserto.TodoApp.Domain.Models;
 using Aserto.TodoApp.Domain.Services;
 using Aserto.TodoApp.Domain.Services.Communication;
 using Aserto.TodoApp.Persistence.Contexts;
+using Aserto.TodoApp.Options;
+using Aserto.AspNetCore.Middleware.Clients.Directory.V3;
+using Aserto.AspNetCore.Middleware.Options;
 
 namespace Aserto.TodoApp.Services
 {
     public class TodoService : ITodoService
     {
         private readonly AppDbContext db;
-
-        public TodoService(AppDbContext dbContext)
+        private readonly Aserto.AspNetCore.Middleware.Clients.Directory.V3.Directory directoryClient;
+        private readonly DirectoryConfig opts;
+        public TodoService(IOptions<DirectoryConfig> config, AppDbContext dbContext)
         {
             db = dbContext;
+            this.opts = config.Value;
+            if (!opts.IsValid())
+            {
+                throw new Exception("Invalid config");
```

Comment:
This isn't a particularly useful error message. Would it make sense to replace `opts.IsValid()` with `opts.Validate()` that throws an exception with a meaningful error message depending on what exactly is invalid?